src/exchange/disclaimer/disclaimer.controller.js

```diff
@@ -28,44 +27,73 @@ angular
-            this.loading.table = true;
 
-            this.services
-                .Exchange.getDisclaimers(this.$routerParams.organization, this.$routerParams.productId, count, offset)
+        refreshList () {
+            if (!this.disclaimersList) { return undefined; }
+            const config = { pageSize: this.pageSize, offset: this.offset };
+            this._getDisclaimers(config)
                 .then((disclaimers) => {
-                    this.disclaimersList = disclaimers;
-                    this.setMessagesFlags(disclaimers);
+                    for (let i = 0; i < disclaimers.list.results.length; i++) {
+                        this.disclaimersList.splice(i, 1, disclaimers.list.results[i]);
+                    }
+                    for (let i = disclaimers.list.results.length; i < this.disclaimersList.length; i++) {
+                        this.disclaimersList.splice(i, 1);
+                    }
+                    this.setMessagesFlags(this.disclaimers);
                 })
                 .catch((err) => {
                     this.services.messaging.writeError(this.services.translator.tr("exchange_tab_DISCLAIMER_error_message"), err);
-                })
-                .finally(() => {
-                    this.loading.table = false;
                 });
         }
 
-        setMessagesFlags (disclaimersList) {
-            const disclaimersHaveEmptylot = this.hasEmptySlot(disclaimersList.list.results);
-            const disclaimersHaveFullSlot = this.hasFullSlot(disclaimersList.list.results);
 
-            this.addDomainMessageFlag = _.isEmpty(disclaimersList.list.results) || (!disclaimersHaveEmptylot && !disclaimersHaveFullSlot);
-            this.noDisclaimerMessageFlag = disclaimersHaveEmptylot && !disclaimersHaveFullSlot;
+        loadPaginated ($config) {
+            this.disclaimersList = null;
+            this.pageSize = $config.pageSize;
+            this.offset = $config.offset;
+            return this._getDisclaimers($config)
+                .then((disclaimers) => {
+                    this.disclaimersList = disclaimers.list.results;
+                    this.setMessagesFlags(this.disclaimers);
+                    return {
+                        data: this.disclaimersList,
+                        meta: {
+                            totalCount: disclaimers.count
+                        }
+                    };
+                })
+                .catch((err) => {
+                    this.services.messaging.writeError(this.services.translator.tr("exchange_tab_DISCLAIMER_error_message"), err);
+                });
+        }
+
+        setMessagesFlags () {
+            this.addDomainMessageFlag = !this.newDisclaimersDisabled();
         }
 
         newDisclaimersDisabled () {
-            return this.disclaimersList != null && !this.hasEmptySlot(this.disclaimersList.list.results);
+            return !this.disclaimersList || !this.hasEmptySlot(this.disclaimersList);
```

Comment:
my way his shorter and don't use a stupid bloated function.

Reply:
_.isObject would be a poor usage here since the function is intended to check if something is an object or not.  If you want to use lodash for that, you'd have to use either _.isNull or _.isUndefined.... but it is not needed since pretty much every javascript dev knows that null and undefined return falsy anyway.